### PR TITLE
Ft-Update-Profile-One-User-UqDcj25i

### PR DESCRIPTION
**Pull request by SIBONIYO**
#### What does this PR do?
- Shows some CRUD operations of One User profile
#### Description of Task to be completed?
- You can GET a profile of One user
- you can DELETE a profile of One user
#### How should this be manually tested?
- This can be tested by using Postman http://localhost:3000/api/users/userId or by 
- swagger http://localhost:3000/api/documentation/#/Users
#### What are the relevant pivotal tracker stories?
[UqDcj25i](https://trello.com/c/UqDcj25i/38-update-profile)

### CODE REVIEW

File: src/middlewares/checkOwnerToken.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { config } from 'dotenv';
+
+config();
+
+export default async (req, res, next) => {
+  try {
+    const token = req.headers.auth.split(' ')[1];
+    const info = jwt.decode(token, process.env.JWT_KEY);
+    const userId = info.id;
+    if (parseInt(req.params.userId) !== userId && req.user.role !== 'admin') {
```
Comment (raymond42) [inline src/middlewares/checkOwnerToken.js:13]: should the operator be `&` or `||`?  if we use `&` that means if one of the statements is not met, the whole function becomes `false` regardless of the other check. So I think we use the `or` operator instead. What do you think?

File: src/middlewares/checkOwnerToken.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      return errorRes(res, 401, res.__('Not authorized'));
```
Comment (raymond42) [inline src/middlewares/checkOwnerToken.js]: Can we change the error message to `You are not allowed to perform this action`

File: src/routers/usersRouter.js [hunk 3]
```diff
@@ -501,7 +503,77 @@ userRouter.get(
...
+ *   responses:
+ *    200:
+ *     description: User deleted successfully
+ *    404:
+ *     description: User not found :(
+ *    500:
+ *     description: Internal server error
+ */
+
+userRouter.delete('/:userId', checktoken, isNotDriver, deleteOne);
```
Comment (raymond42) [inline src/routers/usersRouter.js]: why are we checking `isNotDriver` functionalities? shouldn't it be `checkOwner` instead?

**Review by raymond42 — CHANGES_REQUESTED** [review]
Good job. Attend to the comments and revert to me

**Review by mansurissa — APPROVED** [review]

File: src/controllers/usersController.js [hunk 4]
```diff
@@ -122,6 +122,58 @@ export const getAll = async (req, res) => {
...
+    const { userId } = req.params;
+    const user = await User.findOne({
+      where: { id: userId },
+      attributes: [
+        'id',
+        'firstName',
+        'lastName',
+        'nationalId',
+        'phone',
+        'email'
```
Comment (mansurissa) [inline src/controllers/usersController.js]: I think you didn't need to add attributes cause you wanted to delete

**Review by mansurissa — CHANGES_REQUESTED** [review]

File: src/controllers/usersController.js (hunk 4 above)
Comment (SIBONIYO) [reply src/controllers/usersController.js]: ok let me work on it

**Comment by SIBONIYO** [comment]
let me work on it?

**Review by 41hulk — COMMENTED** [review]
LGTM

File: src/routers/usersRouter.js (hunk 3 above)
Comment (SIBONIYO) [reply src/routers/usersRouter.js]: i think the Owner should not have rights to delete his own profile

File: src/controllers/locales/en.json [hunk 5]
```diff
@@ -148,5 +148,17 @@
...
+	"Successfully deleted a user": "Successfully deleted a user",
+	"Role Driver is not allowed": "Role Driver is not allowed",
+	"Role Operator is not allowed": "Role Operator is not allowed",
+	"You are not allowed. Check Your password": "You are not allowed. Check Your password",
+	"You are not allowed to view the profile": "You are not allowed to view the profile",
+	"You are not authorized, check your Id": "You are not authorized, check your Id",
+	"you are not allowed to view the profile": "you are not allowed to view the profile",
+	"Not authorized. Not owner": "Not authorized. Not owner",
+	"Not authorized": "Not authorized"
 }
```
Comment (KundwaBruno) [inline src/controllers/locales/en.json]: great work, remember to use eslint to add new lines for you

File: src/middlewares/checkOwnerToken.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    }
+    return next();
+  } catch (error) {
+    console.log(error);
```
Comment (KundwaBruno) [inline src/middlewares/checkOwnerToken.js]: and remove these console logs

**Review by KundwaBruno — CHANGES_REQUESTED** [review]

File: src/controllers/usersController.js [hunk 7]
```diff
@@ -144,7 +189,7 @@ export const signin = async (req, res) => {
       }
     });
   } catch (error) {
-    return errorRes(res, 500, res.__('There was error while signining in'));
+    return errorRes(res, 500, res.__(''));
```
Comment (raymond42) [inline src/controllers/usersController.js]: why did you change the error message to an empty string?

File: src/middlewares/checkOwnerToken.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  try {
+    const token = req.headers.auth.split(' ')[1];
+    const info = jwt.decode(token, process.env.JWT_KEY);
+    const userId = info.id;
+    if (parseInt(req.params.userId) !== userId && req.user.role !== 'admin') {
+      return errorRes(res, 401, res.__('Not allowed to perform this action'));
+    }
+    return next();
+  } catch (error) {
+    console.log(error);
```
Comment (raymond42) [inline src/middlewares/checkOwnerToken.js]: remove the console.log

Comment (SIBONIYO) [reply src/middlewares/checkOwnerToken.js]: done working on it

File: src/middlewares/checktoken.js [hunk 9]
```diff
@@ -33,6 +33,7 @@ export default async (req, res, next) => {
       return next();
     });
   } catch (error) {
+    console.log(error);
```
Comment (raymond42) [inline src/middlewares/checktoken.js]: remove this console.log

**Review by raymond42 — APPROVED** [review]
LGTM. Nice work
